wptrunner/executors/pytestrunner/fixtures.py

```diff
@@ -45,14 +46,79 @@ def setup(request, session):
         def test_something(setup, session):
             assert session.url == "https://example.org"
 
-    The session is closed when the test module goes out of scope by an
-    implicit call to `session.end`.
+    When the test function goes out of scope, any remaining user prompts
+    and opened windows are closed, and the current browsing context is
+    switched back to the top-level browsing context.
+
+    After returning the session to a near-pristine state, the session
+    is also closed by an implicit call to `session.end`.
     """
 
     def __init__(self, client):
         self.client = client
 
-    @pytest.fixture(scope="module")
+        # store handle of original window
+        # so that we do not close it
+        self.original_window = self.client.window_handle
+        self.client.end()
```

Comment:
ah, interesting, then should you guard the call to original_window with `if client.session_id` ?  So we aren't just creating a new driver instance and then throwing it away for no good reasion?

and then I don't understand the 'restore_windows' since shouldn't a call to 'end' result in any existing windows being closed?